mm2src/trading_api/src/one_inch_api/client.rs

```diff
@@ -0,0 +1,174 @@
+use super::errors::ApiClientError;
+use crate::one_inch_api::errors::NativeError;
+use common::StatusCode;
+use lazy_static::lazy_static;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::{map_mm_error::MapMmError,
+                     map_to_mm::MapToMmResult,
+                     mm_error::{MmError, MmResult}};
+use mm2_net::transport::slurp_url_with_headers;
+use serde::de::DeserializeOwned;
+use url::Url;
+
+#[cfg(any(test, feature = "mocktopus"))]
+use mocktopus::macros::*;
+
+const ONE_INCH_API_ENDPOINT_V6_0: &str = "swap/v6.0/";
+const SWAP_METHOD: &str = "swap";
+const QUOTE_METHOD: &str = "quote";
+const LIQUIDITY_SOURCES_METHOD: &str = "liquidity-sources";
+const TOKENS_METHOD: &str = "tokens";
+
+const ONE_INCH_AGGREGATION_ROUTER_CONTRACT_V6_0: &str = "0x111111125421ca6dc452d289314280a0f8842a65";
+const ONE_INCH_ETH_SPECIAL_CONTRACT: &str = "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee";
+
+#[cfg(test)]
+const ONE_INCH_API_TEST_URL: &str = "https://api.1inch.dev";
+
+lazy_static! {
+    /// API key for testing
+    static ref ONE_INCH_API_TEST_AUTH: String = std::env::var("ONE_INCH_API_TEST_AUTH").unwrap_or_default();
+}
+
+pub(crate) type QueryParams<'life> = Vec<(&'life str, String)>;
+
+/// 1inch v6.0 supported eth-based chains
+const ONE_INCH_V6_0_SUPPORTED_CHAINS: &[(&str, u64)] = &[
+    ("Ethereum", 1),
+    ("Optimism", 10),
+    ("BSC", 56),
+    ("Gnosis", 100),
+    ("Polygon", 137),
+    ("Fantom", 250),
+    ("ZkSync", 324),
+    ("Klaytn", 8217),
+    ("Base", 8453),
+    ("Arbitrum", 42161),
+    ("Avalanche", 43114),
+    ("Aurora", 1313161554),
+];
```

Comment:
I don't like this in case they add support for new chains, in next PRs you should find a way to make this configurable or look for an API that provides this info.

Reply:
I thought of adding this into a config file (but did not find suitable one).
For the other hand, if the QA team has to add another chain they would still need to make a PR to the KDF, so maybe they could modify this const as well(?)

(Besides, I think if 1inch ever adds a new chain they will probably have to make a new version of their API, so we would need to modify the code)